### PR TITLE
Modify bridge to return the same response when submitting same solution 

### PR DESCRIPTION
**Pull request by julian-berbel**

### CODE REVIEW

File: app/assets/javascripts/application/bridge.js [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
   Laboratory.prototype = {
     runLocalTests: function (solution) {
-      var token = new mumuki.CsrfToken();
-      var request = token.newRequest({
-        type: 'POST',
-        url: window.location.origin + window.location.pathname + '/solutions',
-        data: solution
-      });
-      return $.ajax(request)
+      if(lastSubmission.result && lastSubmission.result.status !== 'aborted' && JSON.stringify(lastSubmission.content) === JSON.stringify(solution)){
```
Comment (fedescarpa) [inline app/assets/javascripts/application/bridge.js]: Esto lo haces para comparar todo el JSON por igualdad, ¿verdad? Con lodash se puede hacer, pero acá no hay lodash. Así que banco esto que hiciste, yo solo lo delegaría para aumentar el intention revealing

**Comment by julian-berbel** [comment]
Since I never cancelled Rails' autosubmission when I made bridge in the first place, it is intervening and auto sending the solution anyways the second time. Before merging this I should find some way to cancel that.

File: app/assets/javascripts/application/bridge.js [hunk 2]
```diff
@@ -1,19 +1,39 @@
...
     runLocalTests: function (solution) {
-      var token = new mumuki.CsrfToken();
-      var request = token.newRequest({
-        type: 'POST',
-        url: window.location.origin + window.location.pathname + '/solutions',
-        data: solution
-      });
-      return $.ajax(request)
+      if(lastSubmissionFinishedSuccessfully() && sameAsLastSolution(solution)){
+        return $.Deferred().resolve(lastSubmission.result)
```
Comment (flbulgarelli) [inline app/assets/javascripts/application/bridge.js]: Missing `;` and missing spaces on `else`.

File: app/assets/javascripts/application/bridge.js [hunk 3]
```diff
@@ -1,19 +1,39 @@
...
+      }else{
+        var token = new mumuki.CsrfToken();
+        var request = token.newRequest({
```
Comment (flbulgarelli) [inline app/assets/javascripts/application/bridge.js]: We should delegate this part of the method to something like `sendNewSolution`

**Pull request merged**